docker/proof-server-chart/templates/proof-server-hpa.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "proof-server.fullname" . }}-hpa
+  labels:
+  {{- include "proof-server.labels" . | nindent 4 }}
+spec:
+  maxReplicas: 5
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 50
+        type: Utilization
+    type: Resource
+  minReplicas: 2
```

Comment:
We may want to make min/max Replicas, and averageUtilization configurable via values.yaml.